### PR TITLE
tests/net: add stub test for gcoap_forward_proxy

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

This adds enough to do compile testing for gcoap_forward_proxy. A proper test would still be needed, though.

This still already exposed compilation issues.

### Testing procedure

The test is a stub to at least get compilation testing. It is not intended to be flashed and run.

So: Green CI

### Issues/PRs references

A step towards fixing https://github.com/RIOT-OS/RIOT/issues/20910

### CODE REVIEW

File: tests/net/gcoap_forward_proxy/Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+CFLAGS += -DTEST_SUITES
+CFLAGS += -DHAS_SOCK_DNS_MOCK=1
+
+include $(RIOTBASE)/Makefile.include
+
+ifneq (,$(filter native native64,$(BOARD)))
+test: PORT=
+$(call target-export-variables,test,PORT)
+endif
```
Comment (miri64) [inline tests/net/gcoap_forward_proxy/Makefile:33]: What is this about?

File: tests/net/gcoap_forward_proxy/Makefile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+ifeq (1,$(COAPS))
+  USEMODULE += gcoap_dtls
+  USEMODULE += prng_sha256prng
+  USEPKG += tinydtls
+  KCONFIG_ADD_CONFIG += $(CURDIR)/coaps.config
+endif
+
+CFLAGS += -DTEST_SUITES
+CFLAGS += -DHAS_SOCK_DNS_MOCK=1
```
Comment (miri64) [inline tests/net/gcoap_forward_proxy/Makefile:26]: Seems like copy-pasta too.

Comment (miri64) [inline tests/net/gcoap_forward_proxy/Makefile:26]: Seems like copy-pasta too.

**Comment by maribu** [comment]
Cleaned up copy-pasta leftovers in the Makefile and squashed

**Review by miri64 — APPROVED** [review]

**Pull request merged**